### PR TITLE
Add KHR_parallel_shader_compile benchmark

### PR DESCRIPTION
**Pull request by jchen10**
Chromium bug: http://crbug.com/849576

### CODE REVIEW

**Comment by gyagp** [comment]
One overall comment is shader loading can be a shared pass for both serial or parallel compile (no need to load twice), and it's better to put all the logic for serial or parallel in a separate function to clearly show their  usage difference.

**Comment by jchen10** [comment]
@gyagp thanks for reviewing. I made a further refactor. PTAL!

**Comment by jdashg** [comment]
This is also going to suffer on first-run when shaders aren't in the browser/systems cache, vs subsequent runs where if may be.
Ideally we'd randomly perturb some shader values to ensure that it's not cached, and allow testing with caching as well if we want. First-run cold-cache tends to be much more important here.

We don't need to fix all this, but we should make notes of these limitations in the readme or the test.

File: sdk/tests/performance/parallel_shader_compile/shaders/aquarium/README [hunk 1]
```diff
@@ -0,0 +1 @@
+http://webglsamples.org/aquarium/aquarium.html
```
[the remark this answers is not in the document]
Comment (jchen10) [reply sdk/tests/performance/parallel_shader_compile/shaders/aquarium/README]: Done.
The license of aquarium seems to be "Creative Commons Attribution 3.0 License".
It looks free according to https://html5up.net/license .
@kenrussell Please have a goalkeeper check on this. Thanks!

**Comment by jchen10** [comment]
@jdashg Thank you very much for the comments. I double checked the cache issue. ANGLE computes the original shader source for the program hash. I though it should work to bypass the cache by inserting a comment with random numbers to the shader source. And it actually works with my TOT chromium. But I just found it doesn't work with Chrome stable, beta. Chrome dev has no such problem too. I haven't figure out the root cause. To make it work for now, I inserted more perturbing lines into the shader source.

File: sdk/tests/performance/parallel_shader_compile/shaders/aquarium/README [hunk 2]
```diff
@@ -0,0 +1,10 @@
+1. Pulled from http://webglsamples.org/aquarium/aquarium.html, the license of which is:
+
+<!DOCTYPE HTML>
+<!--
+  Verti by HTML5 UP
+  html5up.net | @n33co
+  Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
```
Comment (kenrussell) [inline sdk/tests/performance/parallel_shader_compile/shaders/aquarium/README]: I don't think this is the correct license. Gregg Tavares wrote the Aquarium while he was at Google, and the license of aquarium.html is:

 * Copyright 2009, Google Inc.
 * All rights reserved.

Please double-check this. Thanks.

File: sdk/tests/performance/parallel_shader_compile/shaders/aquarium/README (hunk 1 above)
[the remark this answers is not in the document]
Comment (kenrussell) [reply sdk/tests/performance/parallel_shader_compile/shaders/aquarium/README]: Gregg Tavares wrote the Aquarium while he was at Google, and aquarium.html contains the following license at the top:

 * Copyright 2009, Google Inc.
 * All rights reserved.

followed by the standard 3-clause BSD license.

Could you copy/paste the license text here please? That should be sufficient attribution. Thanks.

[the remark this answers is not in the document]
Comment (jchen10) [reply sdk/tests/performance/parallel_shader_compile/shaders/aquarium/README]: @kenrussell thank you for the clarification! The license text copied here.

**Comment by jchen10** [comment]
@kenrussell PTAL

**Pull request merged**